src/components/subscriptionDialog.vue

```diff
@@ -0,0 +1,294 @@
+                maxlength="3"
+                placeholder="CVC"
+                id="cvc"
+                outlined
+                dense
+                class="input"
+                name="cvc"
+              />
+            </v-col>
+
+            <v-col>
+              <v-text-field
+                v-model="expiryMonth"
+                type="number"
+                maxlength="2"
+                label="Expiry Month"
+                placeholder="MM"
+                id="expiryMonth"
+                outlined
+                dense
+                class="input"
+                name="expiryMonth"
+              />
+            </v-col>
+
+            <v-col>
+              <v-text-field
+                v-model="expiryYear"
+                type="number"
+                label="Expiry Year"
+                placeholder="YY"
+                id="expiryYear"
+                maxlength="2"
+                outlined
+                dense
+                class="input"
+                name="expiryYear"
+              />
+            </v-col>
+          </v-row>
+        </v-col>
+      </v-list-item>
+
+      <v-card-actions v-if="!showInputs">
+        <v-btn
+          color="grey"
+          text
+          @click="$store.dispatch('HANDLE_SUBSCRIPTION_DIALOG', false)"
+          data-cy="dismiss-modal"
+        >
+          Subscribe Later
+        </v-btn>
+        <v-btn
+          color="primary"
+          text
+          data-cy="subscribe"
+          @click="showInputs = true"
+        >
+          Subscribe Now
+        </v-btn>
+      </v-card-actions>
+
+      <v-card-actions
+        style="display: flex; justify-content: space-between;"
+        v-else
+      >
+        <v-btn
+          color="grey"
+          data-cy="dismiss-modal"
+          text
+          @click="$store.dispatch('HANDLE_SUBSCRIPTION_DIALOG', false)"
+        >
+          Cancel
+        </v-btn>
+        <v-btn
+          color="primary"
+          text
+          id="confirm-btn"
+          data-cy="confirm-btn"
+          :disabled="!cardNumberIsValid"
+          @click="submitSubscription()"
+          style="display: flex;"
+        >
+          {{ loading ? 'Confirming Details' : 'Confirm Details' }}
+          <v-progress-circular
+            style="padding-left: 40px;"
+            indeterminate
+            v-if="loading"
+            :width="2.5"
+            :size="20"
+            color="primary"
+          />
+        </v-btn>
+      </v-card-actions>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script>
+import {
+  HANDLE_SUBSCRIPTION_DIALOG, HANDLE_EDGE_SYNC_DIALOG, FETCH_USER_SUBSCRIPTION
+} from '@/store/action-types'
+import { mapActions, mapState } from 'vuex'
+const cardValidator = /^(?:4[0-9]{12}(?:[0-9]{3})?|[25][1-7][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})$/
+
+export default {
+  name: 'SubcriptionDialog',
+  data: () => ({
```

Comment:
@vickywane why is this test code still in the PR? Do we want end users to see this fake card data?